gramineproject/gramine
# [LibOS,Pal/Linux-SGX] Generate offsets using an executable

## Description of the changes 

Instead of generating assembly code and parsing it, we compile an executable that outputs desired source (C header file or Python module). As a result, the code is less brittle (no need to account for GCC and Clang assembly format), and some hacks can be removed from Meson. 





## How to test this PR? 

CI should be enough, but you can also look at the generated files (`asm-offsets.h` and `_offsets.py`) in the build directory.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/139)
